extractor/src/main/java/com/example/extractor/TemplateProcessor.java

```diff
@@ -0,0 +1,18 @@
+package com.example.extractor;
+
+public class TemplateProcessor {
+
+    public static String replacePlaceholders(String template, String moduleContent, String propertiesContent) {
+        // Replace the treewalker_module_PLACEHOLDER with the actual module content
+        template = template.replace("<!-- treewalker_module_PLACEHOLDER -->", moduleContent);
+
+        // Replace the properties_PLACEHOLDER with the actual properties or remove the line if empty
+        if (propertiesContent.isEmpty()) {
+            template = template.replaceAll("(?m)^\\s*<!-- properties_PLACEHOLDER -->\\s*$", "");
```

Comment:
Module placeholder should contains properties.
No extra placeholder.

Reply:
Done